numpy/core/src/multiarray/convert_datatype.c

```diff
@@ -897,22 +897,6 @@ promote_types(PyArray_Descr *type1, PyArray_Descr *type2,
 
 }
 
-/*
- * Returns a new reference to type if it is already NBO, otherwise
- * returns a copy converted to NBO.
- */
-NPY_NO_EXPORT PyArray_Descr *
-ensure_dtype_nbo(PyArray_Descr *type)
-{
-    if (PyArray_ISNBO(type->byteorder)) {
-        Py_INCREF(type);
-        return type;
-    }
-    else {
-        return PyArray_DescrNewByteorder(type, NPY_NATIVE);
-    }
-}
-
```

Comment:
This is no longer needed for non-structured arrays?

Reply:
It is just moved, but maybe I should limit the change a bit (since this is perfectly fine for most NumPy builtin dtypes).  In any case, it is actually a change of one of the previous PRs that would have to go in first.

Reply:
See #19345